_delphi_utils_python/tests/test_archive.py

```diff
@@ -22,125 +22,199 @@
+            "sample_size": [10.0],
+            "missing_val": [Nans.NOT_MISSING],
+            "missing_se": [Nans.NOT_MISSING],
+            "missing_sample_size": [Nans.NOT_MISSING],
+        }),
+        after=None,
+        diff=None
+    ),
+    "add_file": Example( # was: csv3
+        before=None,
+        after=pd.DataFrame({
+            "geo_id": ["2"],
+            "val": [2.0000002],
+            "se": [0.2],
+            "sample_size": [20.0],
+            "missing_val": [Nans.NOT_MISSING],
+            "missing_se": [Nans.NOT_MISSING],
+            "missing_sample_size": [Nans.NOT_MISSING],
+        }),
+        diff=None
+    ),
+    "unchanged_old_new": Example( # was: csv4
+        before=pd.DataFrame({
+            "geo_id": ["1"],
+            "val": [1.0],
+            "se": [0.1],
+            "sample_size": [10.0]
+        }),
+        after=pd.DataFrame({
+            "geo_id": ["1"],
+            "val": [1.0],
+            "se": [0.1],
+            "sample_size": [10.0],
+            "missing_val": [Nans.NOT_MISSING],
+            "missing_se": [Nans.NOT_MISSING],
+            "missing_sample_size": [Nans.NOT_MISSING],
+        }),
+        diff=pd.DataFrame({
+            "geo_id": ["1"],
+            "val": [1.0],
+            "se": [0.1],
+            "sample_size": [10.0],
+            "missing_val": [Nans.NOT_MISSING],
+            "missing_se": [Nans.NOT_MISSING],
+            "missing_sample_size": [Nans.NOT_MISSING],
+        })
+    ),
+    "mod_2_del_3_add_4_old_old": Example( # was: csv5
+        before=pd.DataFrame({
+            "geo_id": ["1", "2", "3"],
+            "val": [1.0, 2.0, 3.0],
+            "se": [np.nan, 0.20000002, 0.30000003],
+            "sample_size": [10.0, 20.0, 30.0]
+        }),
+        after=pd.DataFrame({
+            "geo_id": ["1", "2", "4"],
+            "val": [1.0, 2.1, 4.0],
+            "se": [np.nan, 0.21, np.nan],
+            "sample_size": [10.0, 21.0, 40.0]
+        }),
+        diff=pd.DataFrame({
+            "geo_id": ["2", "3", "4"],
+            "val": [2.1, np.nan, 4.0],
+            "se": [0.21, np.nan, np.nan],
+            "sample_size": [21.0, np.nan, 40.0],
+            "missing_val": [np.nan, Nans.DELETED, np.nan],
+            "missing_se": [np.nan, Nans.DELETED, np.nan],
+            "missing_sample_size": [np.nan, Nans.DELETED, np.nan],
+        })
+    )
 }
+EXPECTEDS = Expecteds(
+    deleted=["delete_file.csv"],
+    common_diffs=dict(
+        (f"{csv_name}.csv", None if dfs.diff is EMPTY else f"{csv_name}.csv.diff")
+        for csv_name, dfs in CSVS.items() if dfs.diff is not None
+    ),
+    new=["add_file.csv"],
+)
+# check for incomplete modifications to tests
+assert set(EXPECTEDS.new) == set(f"{csv_name}.csv" for csv_name, dfs in CSVS.items() if dfs.before is None), \
+    "Bad programmer: added more new files to CSVS.after without updating EXPECTEDS.new"
+EXPECTEDS.raw_exports = list(EXPECTEDS.common_diffs.keys()) + \
```

Comment:
`raw_exports`, `diffed_exports`, and `filtered_exports` can probably be put into the `__post_init__` of `Expecteds`, since they're all derived from the inputs.

Reply:
`raw_exports` appears to be publicly unused, maybe just a local variable in `post_init`?

Reply:
hmmm i could go either way -- it's largely here as a building block for `EXPECTEDS.diffed_exports`.

pros:
* it makes the 3 `EXPECTEDS.*_exports` mirror the 3 phases of archive-diffing
* it makes the `EXPECTEDS.diffed_exports` definition (raw exports + .diff files), which matches the (my?) mental model of execution better than (common_diffs.keys() + new + .diff files)

cons:
* not actually used in tests
* adds clutter / cognitive load

i don't feel strongly about it though so if cons outweigh pros for you i can make the change

Reply:
also TIL `__post_init__` !!

Reply:
Yea, doesn't seem like a big deal. I'll make the `__post_init__` suggestion above.